### PR TITLE
Hide docks when you put data modeler in background

### PR DESCRIPTION
**Pull request by ederign**
Depends on: uberfire/uberfire#166

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 1]
```diff
@@ -225,6 +211,19 @@ public void onTextChange() {
     public void onFocus() {
         if ( !loading && context != null ) {
             dataModelerWBContext.setActiveContext( context );
+            enableDataModelerDocksWhenIsHidden();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: Sorry.. being pedantic.. in the ```onFocus``` handler you call a method ```enableDataModelerDocksWhenIsHidden()```... would this second method be best called something like ```showDataModellerDocks()```?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 2]
```diff
@@ -225,6 +211,19 @@ public void onTextChange() {
...
+        }
+    }
+
+    private void enableDataModelerDocksWhenIsHidden() {
+        dataModelerFocusEvent.fire( new DataModelerWorkbenchFocusEvent() );
+    }
+
+    public void disableDataModelerDocksWhenEditorIsHidden( @Observes PlaceHidEvent event ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: Likewise would this be better called ```hideDataModellerDocks```?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 3]
```diff
@@ -225,6 +211,19 @@ public void onTextChange() {
     public void onFocus() {
         if ( !loading && context != null ) {
             dataModelerWBContext.setActiveContext( context );
+            showDataModellerDocks();
+        }
+    }
+
+    private void showDataModellerDocks() {
```
Comment (ederign) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: @manstis

**Comment by manstis** [comment]
*kie-wb-common*
(master) http://github.com/droolsjbpm/kie-wb-common/commit/ab8ebba04

(6.3.x) http://github.com/droolsjbpm/kie-wb-common/commit/4c6190ebc

**Comment by manstis** [comment]
Thanks @ederign

**Pull request closed without merging**